src/main/scala/io/iohk/ethereum/network/PeerEventBusActor.scala

```diff
@@ -0,0 +1,131 @@
+    case class MessageSentClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
+    case class PeerDisconnectedClassifier(peerSelector: PeerSelector) extends SubscriptionClassifier
+    case object PeerHandshaked extends SubscriptionClassifier
+  }
+
+  sealed trait PeerEvent
+
+  object PeerEvent{
+    case class MessageFromPeer(message: Message, peerId: PeerId) extends PeerEvent
+    case class MessageToPeer(message: Message, peerId: PeerId) extends PeerEvent
+    case class PeerDisconnected(peerId: PeerId) extends PeerEvent
+    case class PeerHandshakeSuccessful[R <: HandshakeResult](peer: Peer, handshakeResult: R) extends PeerEvent
+  }
+
+  case class Subscription(subscriber: ActorRef, classifier: SubscriptionClassifier)
+
+  class PeerEventBus extends ActorEventBus {
+
+    override type Event = PeerEvent
+    override type Classifier = SubscriptionClassifier
+
+    private var subscriptions: Seq[Subscription] = Nil
+
+    override def subscribe(subscriber: ActorRef, to: Classifier): Boolean = {
+      val subscription = Subscription(subscriber, to)
+      if (subscriptions.contains(subscription)) {
+        false
+      } else {
+        subscriptions = subscriptions :+ subscription
+        true
+      }
+    }
+
+    override def unsubscribe(subscriber: ActorRef, from: Classifier): Boolean = {
+      val subscription = Subscription(subscriber, from)
+      if (subscriptions.contains(subscription)) {
+        subscriptions = subscriptions.filterNot(_ == subscription)
+        true
+      } else {
+        false
+      }
+    }
+
+    override def unsubscribe(subscriber: ActorRef): Unit = {
+      subscriptions = subscriptions.filterNot(_.subscriber == subscriber)
+    }
+
+    override def publish(event: PeerEvent): Unit = {
+      val interestedSubscribers = event match {
+        case MessageFromPeer(message, peerId) =>
+          subscriptions.collect {
+            case Subscription(subscriber, classifier: MessageReceivedClassifier)
+              if classifier.peerSelector.contains(peerId) &&
+                classifier.messageCodes.contains(message.code) => subscriber
+          }
+        case MessageToPeer(message, peerId) =>
+          subscriptions.collect {
+            case Subscription(subscriber, classifier: MessageSentClassifier)
+              if classifier.peerSelector.contains(peerId) &&
+                classifier.messageCodes.contains(message.code) => subscriber
+          }
+        case PeerDisconnected(peerId) =>
+          subscriptions.collect {
+            case Subscription(subscriber, classifier: PeerDisconnectedClassifier)
+              if classifier.peerSelector.contains(peerId) => subscriber
+          }
+        case _: PeerHandshakeSuccessful[_] =>
+          subscriptions.collect {
+            case Subscription(subscriber, PeerHandshaked) => subscriber
+          }
+      }
```

Comment:
Not yet, @AlanVerbner's changes should get merged soon (#204) and I'll incorporate them into this PR then